test/validator/samples/assignment-to-const-7/input.svelte

```diff
@@ -0,0 +1,6 @@
+<script>
+	const arr = [{1: {arr: 2}}, 2];
+	[{1: { arr }}, arr[1]] = [arr[1], arr[0]];
```

Comment:
I think this is incorrect JavaScript syntax anyway.
(At least I had an error on TS playground.)

<img width="638" alt="image" src="https://user-images.githubusercontent.com/19153718/205471341-514adac0-a322-4f34-a78a-60fa00fed512.png">

Reply:
Yes. This test exactly checks this point.